client_end_user/shared/src/commonMain/kotlin/presentation/auth/signup/registrationSubmit/RegistrationSubmitScreenModel.kt

```diff
@@ -60,21 +57,21 @@ class RegistrationSubmitScreenModel(
         tryCatch { validation.validateFullName(fullName); clearErrors() }
     }
 
-    override fun onEmailChanged(email: String) {
-        updateState { it.copy(email = email) }
-        tryCatch { validation.validateEmail(email); clearErrors() }
-    }
-
     override fun onPhoneChanged(phone: String) {
         updateState { it.copy(phone = phone) }
         tryCatch { validation.validatePhone(phone,_language.value.value); clearErrors() }
     }
 
+    override fun onAddressChanged(address: String) {
+        updateState { it.copy(address = address) }
+        tryCatch { validation.validateAddress(address); clearErrors() }
+    }
+
```

Comment:
could add clearErrors() to the function tryCatch instead of invoking it many times
if the function tryCatch is built in the language make another function that uses it and invokes  clearErrors() at the end